acs-integration-tests/src/test/java/com/ge/predix/test/utils/ACSRestTemplateFactory.java

```diff
@@ -154,12 +156,26 @@ public OAuth2RestTemplate getOAuth2ResttemplateForZAC() {
         return this.zacTemplate;
     }
 
+    // TODO Move the UAA template gets to UAATestUtil so that the ACSRestTemplateFactory is only for ACSRestTemplates
     public OAuth2RestTemplate getOAuth2RestTemplateForUaaAdmin() {
         if (this.uaaAdminTemplate == null) {
             ClientCredentialsResourceDetails resource = new ClientCredentialsResourceDetails();
             resource.setAccessTokenUri(this.uaaTokenUrl);
             resource.setClientId("admin");
-            resource.setClientSecret("adminsecret");
+            resource.setClientSecret(this.uaaAdminSecret);
+            this.uaaAdminTemplate = new OAuth2RestTemplate(resource);
+            this.uaaAdminTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+        }
+
+        return this.uaaAdminTemplate;
+    }
+
+    public OAuth2RestTemplate getOAuth2RestTemplateForUaaAdmin(final String tokenUri) {
```

Comment:
We can start with this method though.  Pass client id and secret as parameters.  Eventually, every method will be changed to not use wired values,